src/community/ogcapi/ogcapi-core/src/main/java/org/geoserver/ogcapi/MessageConverterResponseAdapter.java

```diff
@@ -106,8 +106,13 @@ protected Operation getOperation(T result, Request dr, MediaType mediaType) {
                 .stream()
                 .filter(
                         r ->
-                                getMediaTypeStream(r).anyMatch(mt -> mediaType.isCompatibleWith(mt))
-                                        || (r.canHandle(op) && r.getBinding().isInstance(result)))
+                                r.canHandle(op)
+                                        && (getMediaTypeStream(r)
+                                                        .anyMatch(
+                                                                mt ->
```

Comment:
This stream chain became hard to read, please refactor the nested filter condition in a separated private method and add Javadoc comment about what it does.

Reply:
fixed